File: tools_box/_maintenance/report/generator_fuel_consumption_log_report/generator_fuel_consumption_log_report.js
// Copyright (c) 2016, masonarmani38@gmail.com and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Generator Fuel Consumption Log Report"] = {
	"filters": [{
			"fieldname":"modified_from",
			"label": __("Modified From"),
			"fieldtype": "Datetime",
			"width": "80",
			"reqd":1,
			"default":dateutil.year_start()
		},
		{
			"fieldname":"modified_to",
			"label": __("Modified To"),
			"fieldtype": "Datetime",
			"width": "80",
			"reqd":1,
			"default":dateutil.year_end()
		},
		{
			"fieldname":"generator",
			"label": __("Generator"),
			"fieldtype": "Link",
			"options": "Asset",
			"reqd":0,
		},
	]
}

File: tools_box/_maintenance/report/generator_fuel_consumption_log_report/generator_fuel_consumption_log_report.py
# Copyright (c) 2013, masonarmani38@gmail.com and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe

def execute(filters=None):
	columns = [
		"Name:Link/Generator Fuel Consumption Log:100",
		"Date:Date:120",
		"Generator:Link/Asset:150",
		"Generator Name:Data:250",
		"Fuel Qty:Float:150",
		"Unit Price:Currency/currency;100",
		"Total Price:Currency/currency:100",
	]
	data, conditions = [], ""

	if filters.get('modified_from') and filters.get('modified_to'):
		conditions = " and p.date between DATE('{modified_from}') and DATE('{modified_to}')"

	if filters.get('generator'):
		conditions += " and p.generator = '{generator}'"

	data = frappe.db.sql("SELECT p.name , p.date, p.generator, p.generator_name, c.fuel_qty, c.unit_price, c.total "
							"FROM `tabGenerator Fuel Consumption Log` p JOIN `tabGenerator Fuel Consumption Log Item` c "
							"ON (p.name = c.parent) WHERE (1=1){conds}".format(conds=conditions.format(**filters)),
						 as_list=1)
	return columns, data
